server/src/main/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverAction.java

```diff
@@ -333,4 +256,125 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
             );
         }
     }
+
+    class RolloverTask implements ClusterStateTaskListener {
+
+        private final RolloverRequest rolloverRequest;
+        private final IndicesStatsResponse statsResponse;
+        private final RolloverResponse trialRolloverResponse;
+        private final ActionListener<RolloverResponse> listener;
+
+        private final AtomicBoolean conditionsMet = new AtomicBoolean(false);
```

Comment:
Could this not be a plain boolean?

Reply:
Also, I wonder if `changed` or `performedRollover` were better names - since this is used to figure out if anything changed when the `clusterStateProcessed` method is called?

Reply:
I think this can be a boolean, but it should be volatile? I'm not 100% sure whether performRollover() method is executed by the same thread as clusterStateProcessed() method.

I like the name `clusterStateProcessed` and will change it to that.

Reply:
>  I'm not 100% sure whether performRollover() method is executed by the same thread as clusterStateProcessed() method.

I think it is executed by the same thread. I will change it to be an ordinary boolean.